server/src/Rooms.ts

```diff
@@ -0,0 +1,39 @@
+import {Room} from "./Room";
+import {Player} from "./Player";
+
+export class Rooms {
+
+    rooms :Room[] = [];
+
+    constructor() {}
+
+    addRoom(room: Room) {
+        this.rooms.push(room)
+    }
+
+    deleteRoomByFirstPlayerId(playerId : string) {
+
+        let newRooms = new Rooms()
+
+        for (const room of this.rooms) {
+            if (room.firstPlayerId != playerId) {
+                newRooms.addRoom(room)
+            }
+        }
+
+        this.rooms = newRooms.rooms
+    }
+
+
+    getRoomByCode(code: string) :Room {
```

Comment:
You can use ts-results feature ? https://github.com/vultix/ts-results but it's your call